messaging/src/main/java/org/axonframework/configuration/TypeReference.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.configuration;
+
+import jakarta.annotation.Nonnull;
+
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+
+/**
+ * Represents a reference to a type of component, allowing for generic types to be specified without casting errors.
+ *
+ * @param <E> The type of the component.
+ */
+public abstract class TypeReference<E> {
+    protected final Class<E> type;
```

Comment:
I'd make type private.
Also, wouldn't we want to keep the original Type as well? I'd rename this field to rawType.

We're going to need this class for the converters as well. Might be best to move it to a common/utils package.

Reply:
What Jackson does is throw an error if it's a class without type arguments. This does not seem fair to me.
Then and put the class directly in this `Type` field. Maybe because the carrying type can also have a generic? 
Anyway, I am not biased. I will change it to hold the raw type. 

That means two getters; one to get it as class, and one to get it as type.

The `Class` variant erases any additional generic information, but is needed for component definition.
The `Type` is perhaps needed for serialization.

Reply:
Alright, pushed the change.
The constructor I can not make private, as that would mean it's not constructable. 
The variable is now private

Reply:
Also moved to the common package now.